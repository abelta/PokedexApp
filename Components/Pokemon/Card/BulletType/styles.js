import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  bulletType: {
    padding: 2,
    flex: 1,
  },
  value: {
    color: 'white',
    flex: 1,
    textAlign: 'center',
  },
});

export default styles;
