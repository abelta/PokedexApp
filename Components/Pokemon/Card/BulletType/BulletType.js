import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { typeColor } from '../../../../Helpers/Pokemon';
import styles from './styles';

const BulletType = ({ style, value }) => (
  <View style={[style, styles.bulletType, { backgroundColor: typeColor(value) }]}>
    <Text style={styles.value}>{value}</Text>
  </View>
);

BulletType.propTypes = {
  // eslint-disable-next-line react/require-default-props
  style: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  value: PropTypes.string.isRequired,
};

export default BulletType;
