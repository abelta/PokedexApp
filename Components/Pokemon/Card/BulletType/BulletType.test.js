import React from 'react';
import renderer from 'react-test-renderer';
import BulletType from '.';

describe('BulletType', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BulletType value="HP" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
