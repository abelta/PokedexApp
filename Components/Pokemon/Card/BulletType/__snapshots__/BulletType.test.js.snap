// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletType renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "black",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "flex": 1,
        "textAlign": "center",
      }
    }
  >
    HP
  </Text>
</View>
`;
